src/applications/disability-benefits/all-claims/config/submitForm.js

```diff
@@ -37,31 +41,51 @@ const submitFormFor = eventName =>
             error = new Error(`vets_server_error: ${req.statusText}`);
           }
           error.statusText = req.statusText;
+          clearTimeout(timer);
           reject(error);
         }
       });
 
       req.addEventListener('error', () => {
         const error = new Error('client_error: Network request failed');
         error.statusText = req.statusText;
+        clearTimeout(timer);
         reject(error);
       });
 
       req.addEventListener('abort', () => {
         const error = new Error('client_error: Request aborted');
         error.statusText = req.statusText;
+        clearTimeout(timer);
         reject(error);
       });
 
       req.addEventListener('timeout', () => {
         const error = new Error('client_error: Request timed out');
         error.statusText = req.statusText;
+        clearTimeout(timer);
         reject(error);
       });
 
       req.setRequestHeader('X-Key-Inflection', 'camel');
       req.setRequestHeader('Content-Type', 'application/json');
 
+      // Throw an error after 30 seconds
+      timer = setTimeout(() => {
+        const error = new Error('client_error: Request taking too long');
+        recordEvent({ event: `${eventName}--submission-taking-too-long` });
+        Sentry.withScope(scope => {
+          scope.setExtra('XMLHttpRequest', req);
+          // Too much PII for sentry?
+          // scope.setExtra('form data', form);
+          // scope.setExtra('request body', body);
+          Sentry.captureMessage('Form 526: submission request taking too long');
+        });
+        // eslint-disable-next-line no-console
+        console.log(req, form, JSON.parse(body));
```

Comment:
Is there a reason why we are logging things to the console?

Reply:
In the rare case where the submit button "sticks" and the veteran is unable to proceed, this timer will break out of the promise and log their PII. This is only in case we're in a call with the veteran and need to troubleshoot the problem. I wasn't sure if including the `body` in the Sentry log was appropriate.

Reply:
Also, I'm not sure how I'd write a unit test for this case. We would have to block the promise to get the timer to fire.

Reply:
Linking the [slack thread](https://dsva.slack.com/archives/C5HP4GN3F/p1579627398017700) for posterity.  If the test ends up not being feasible then that's ok, but it might be worth a try.

Reply:
Please take a look at my updates.